### DISCUSSION TITLE
methods doesnt get from server side to flutter side

### DISCUSSION THREAD

**Discussion by kazerdira** [serverpod/serverpod#535] [Q&A]
i tried to write a make a dummy function in the server
Future<Article> getArticle(Session session, int id) async {
    return Article(
        title: 'Flutter with serverPod',
        content: 'this is a test for the project',
        publishedOn: DateTime.now(),
        isPrime: true);
  }
 i tried
 docker-compose up --build --detach
dart bin/main.dart

 but flutter doesnt recognize it when i write 
final testResult = await client.getArticle(2);

and i am getting infinite loop like this 

2022-12-30 21:19:40.295758Z Internal server error. Failed to connect to database in future call manager.
PostgreSQLSeverity.error 42P01: relation "serverpod_future_call" does not exist
package:postgres/src/connection.dart 513:18                     _PostgreSQLExecutionContextMixin._query
package:postgres/src/connection.dart 474:7                      _PostgreSQLExecutionContextMixin.query
package:postgres/src/connection.dart 542:22                     _PostgreSQLExecutionContextMixin.mappedResultsQuery
package:postgres_pool/postgres_pool.dart 769:23                 _PgExecutionContextWrapper.mappedResultsQuery.<fn>
package:postgres_pool/postgres_pool.dart 695:27                 _PgExecutionContextWrapper._run
package:postgres_pool/postgres_pool.dart 768:12                 _PgExecutionContextWrapper.mappedResultsQuery
package:postgres_pool/postgres_pool.dart 631:16                 PgPool.mappedResultsQuery.<fn>
package:postgres_pool/postgres_pool.dart 332:22                 PgPool.run.<fn>.<fn>
package:postgres_pool/postgres_pool.dart 431:27                 PgPool._useOrCreate
package:postgres_pool/postgres_pool.dart 403:14                 PgPool._withConnection.<fn>
package:executor/src/executor_impl.dart 61:19                   _Executor.scheduleTask
package:postgres_pool/postgres_pool.dart 331:18                 PgPool.run.<fn>
package:retry/retry.dart 131:16                                 RetryOptions.retry
package:postgres_pool/postgres_pool.dart 329:14                 PgPool.run
package:serverpod/src/database/database_connection.dart 443:20  DatabaseConnection.deleteAndReturn
package:serverpod/src/database/database.dart 166:12             Database.deleteAndReturn
package:serverpod/src/server/future_call_manager.dart 100:18    FutureCallManager._checkQueue
===== asynchronous gap ===========================
package:postgres_pool/postgres_pool.dart 331:18                 PgPool.run.<fn>
package:retry/retry.dart 131:16                                 RetryOptions.retry
package:postgres_pool/postgres_pool.dart 329:14                 PgPool.run
package:serverpod/src/database/database_connection.dart 443:20  DatabaseConnection.deleteAndReturn
package:serverpod/src/database/database.dart 166:12             Database.deleteAndReturn
package:serverpod/src/server/future_call_manager.dart 100:18    FutureCallManager._checkQueue

Local stacktrace:
#0      FutureCallManager._checkQueue (package:serverpod/src/server/future_call_manager.dart:136:36)
<asynchronous suspension>

2022-12-30 21:19:45.405992Z Internal server error. Failed to connect to database in future call manager.
PostgreSQLSeverity.error 42P01: relation "serverpod_future_call" does not exist
package:postgres/src/connection.dart 513:18                     _PostgreSQLExecutionContextMixin._query
package:postgres/src/connection.dart 474:7                      _PostgreSQLExecutionContextMixin.query
package:postgres/src/connection.dart 542:22                     _PostgreSQLExecutionContextMixin.mappedResultsQuery
package:postgres_pool/postgres_pool.dart 769:23                 _PgExecutionContextWrapper.mappedResultsQuery.<fn>
package:postgres_pool/postgres_pool.dart 695:27                 _PgExecutionContextWrapper._run
package:postgres_pool/postgres_pool.dart 768:12                 _PgExecutionContextWrapper.mappedResultsQuery
package:postgres_pool/postgres_pool.dart 631:16                 PgPool.mappedResultsQuery.<fn>
package:postgres_pool/postgres_pool.dart 332:22                 PgPool.run.<fn>.<fn>
package:postgres_pool/postgres_pool.dart 431:27                 PgPool._useOrCreate
package:postgres_pool/postgres_pool.dart 403:14                 PgPool._withConnection.<fn>
package:executor/src/executor_impl.dart 61:19                   _Executor.scheduleTask
package:postgres_pool/postgres_pool.dart 331:18                 PgPool.run.<fn>
package:retry/retry.dart 131:16                                 RetryOptions.retry
package:postgres_pool/postgres_pool.dart 329:14                 PgPool.run
package:serverpod/src/database/database_connection.dart 443:20  DatabaseConnection.deleteAndReturn
package:serverpod/src/database/database.dart 166:12             Database.deleteAndReturn
package:serverpod/src/server/future_call_manager.dart 100:18    FutureCallManager._checkQueue
===== asynchronous gap ===========================
package:postgres_pool/postgres_pool.dart 331:18                 PgPool.run.<fn>
package:retry/retry.dart 131:16                                 RetryOptions.retry
package:postgres_pool/postgres_pool.dart 329:14                 PgPool.run
package:serverpod/src/database/database_connection.dart 443:20  DatabaseConnection.deleteAndReturn
package:serverpod/src/database/database.dart 166:12             Database.deleteAndReturn
package:serverpod/src/server/future_call_manager.dart 100:18    FutureCallManager._checkQueue

Local stacktrace:
#0      FutureCallManager._checkQueue (package:serverpod/src/server/future_call_manager.dart:136:36)
<asynchronous suspension>

**Comment by kazerdira** [comment]
i deleted everything and start again , so in my cmd console i had those commands


All setup. You are ready to rock!

Start your Serverpod by running:

  $ cd .\demopod\demopod_server\
  $ .\setup-tables.cmd
  $ docker-compose up --build --detach
  $ dart .\bin\main.dart


but when I run second command I mean  .\setup-tables.cmd , i am getting this 



Starting docker
[+] Running 2/2
 - Container demopod_server-redis-1     Started                                                                                                                    4.3s
 - Container demopod_server-postgres-1  Started                                                                                                                    4.5s
Waiting for Postgres...
Postgres is ready
failed to get console mode for stdin: Descripteur non valide
failed to get console mode for stdin: Descripteur non valide
psql: error: connection to server at "postgres" (172.18.0.3), port 5432 failed: FATAL:  password authentication failed for user "postgres"
Stopping docker
[+] Running 2/2
 - Container demopod_server-redis-1     Stopped                                                                                                                    0.8s
 - Container demopod_server-postgres-1  Stopped
 

Just tell me guys please if is possible , is serverpod working on windows 10 or not yet ..if not just tell me what wrong here

**Reply by ajay-k07** [reply]
its say password authentication failed , can you create a new serverpod project  and follow these commands 

```

serverpod create mypod
cd mypod/mypod_server
docker-compose up --build --detach
dart bin/main.dart

```

**Reply by kazerdira** [reply]
still doesnt work , i mean it give the error that i reported in the first place , and can you explain what authentification failed? to docker ?

**Comment by kazerdira** [comment]
Weirdly flutter project recognize my functions from server project , so it is working 

ok , I tried to search I find out i forget to VALIDATE MY DOCKER ACCOUNT AFTER SIGNING IN 

but still having error reports , after running .\setup-tables.cmd

i am getting

Starting docker
[+] Running 21/21
 - redis Pulled                                                                                                                                                  120.0s
   - 1fe172e4850f Pull complete                                                                                                                                  115.6s
   - 6fbcd347bf99 Pull complete                                                                                                                                  115.7s
   - 993114c67627 Pull complete                                                                                                                                  115.9s
   - 2a560260ca39 Pull complete                                                                                                                                  116.5s
   - b7179886a292 Pull complete                                                                                                                                  116.6s
   - 8901ffe2be84 Pull complete                                                                                                                                  116.7s
 - postgres Pulled                                                                                                                                               162.8s
   - 5eb5b503b376 Pull complete                                                                                                                                   74.9s
   - daa0467a6c48 Pull complete                                                                                                                                   75.1s
   - 7cf625de49ef Pull complete                                                                                                                                   75.2s
   - bb8afcc973b2 Pull complete                                                                                                                                   75.3s
   - c74bf40d29ee Pull complete                                                                                                                                   75.7s
   - 2ceaf201bb22 Pull complete                                                                                                                                   77.9s
   - 1255f255c0eb Pull complete                                                                                                                                   78.0s
   - d27501cd0cca Pull complete                                                                                                                                   78.4s
   - ff5b6d09a5d0 Pull complete                                                                                                                                  159.1s
   - f635aec27645 Pull complete                                                                                                                                  159.2s
   - a165c6729250 Pull complete                                                                                                                                  159.3s
   - b0aa4f86b611 Pull complete                                                                                                                                  159.3s
   - 9efc4664d9d2 Pull complete                                                                                                                                  159.4s
[+] Running 4/4
 - Network mypod_server_default       Created                                                                                                                      0.7s
 - Volume "mypod_server_mypod_data"   Created                                                                                                                      0.0s
 - Container mypod_server-redis-1     Started                                                                                                                      3.1s
 - Container mypod_server-postgres-1  Started                                                                                                                      2.8s
Waiting for Postgres...
Postgres is ready
failed to get console mode for stdin: Descripteur non valide
failed to get console mode for stdin: Descripteur non valide
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE TABLE
ALTER TABLE
CREATE TABLE
ALTER TABLE
CREATE INDEX
CREATE INDEX
CREATE INDEX
CREATE TABLE
ALTER TABLE
CREATE INDEX
ALTER TABLE
CREATE TABLE
ALTER TABLE
ALTER TABLE
CREATE TABLE
ALTER TABLE
CREATE INDEX
ALTER TABLE
Stopping docker
[+] Running 2/2
 - Container mypod_server-redis-1     Stopped                                                                                                                      0.8s
 - Container mypod_server-postgres-1  Stopped                                                                                                                      0.7s



then i run docker-compose up --build --detach

i am getting 

SERVERPOD version: 0.9.21 mode: development time: 2022-12-31 19:16:04.687210Z
Insights listening on port 8081
Server default listening on port 8080
Webserver listening on port 8082
CPU and memory usage metrics are not supported on this platform.

**Reply by yahu1031** [reply]
> CPU and memory usage metrics are not supported on this platform.

Currently, this won't be supported in windows.

**Reply by kazerdira** [reply]
But it is working it is may be a magic

**Reply by yahu1031** [reply]
eh? Nope. Cause the plugin wasn't written for windows.

**Reply by yahu1031** [reply]
@kazerdira Checkout here [system_resources](https://pub.dev/packages/system_resources#windows)

**Reply by vlidholt** [reply]
CPU and memory usage metrics isn't really useful when running on your local machine anyway. When you deploy your server, it's recommended that you run it on Linux.
